storage/src/test/scala/coop/rchain/storage/TestTools.scala

```diff
@@ -46,19 +49,21 @@ object TestTools {
     blobs
   }
 
-  def ArraysEqual(a1: Array[String], a2: Array[String]): Boolean = {
-    if (a1.length != a2.length) return false
-    for (i <- 0 until a1.length) {
-      var contains = false
-      for (j <- 0 until a2.length) {
-        if (a1(i) == a2(j)) {
-          contains = true
+  def arraysEqual(a1: Array[String], a2: Array[String]): Boolean = {
+    if (a1.length != a2.length) {
+      false
+    } else {
+      var equal = true
+      val a1sorted = a1.sorted
+      val a2sorted = a2.sorted
+      var i = 0
+      while (i < a1sorted.length && equal) {
```

Comment:
Is there a way sameElements can fail here?  I don't think https://stackoverflow.com/a/29008501 applies to a sorted array.

Reply:
$ scala
Welcome to Scala 2.12.2 (Java HotSpot(TM) 64-Bit Server VM, Java 1.8.0_131).

scala> val a = Array("1", "2")
a: Array[String] = Array(1, 2)

scala> val b = Array("2", "1")
b: Array[String] = Array(2, 1)

scala> a.sameElements(b)
res0: Boolean = false

scala> val x = Array("1", "2", "3", "4", "5")
x: Array[String] = Array(1, 2, 3, 4, 5)

scala> val y = Array("2", "1", "3", "4", "5")
y: Array[String] = Array(2, 1, 3, 4, 5)

scala> x.sameElements(y)
res1: Boolean = false

scala> val i = Array(1, 2)
i: Array[Int] = Array(1, 2)

scala> val j = Array(2, 1)
j: Array[Int] = Array(2, 1)

scala> i.sameElements(j)
res2: Boolean = false